exec/java-exec/src/main/java/org/apache/drill/exec/ops/QueryContext.java

```diff
@@ -283,4 +286,18 @@ public void close() throws Exception {
       closed = true;
     }
   }
+
+  /**
+  * @param stmtType : The type {@link StatementType} e.g. CTAS, ANALYZE of the statement
+  */
+  public void setStatementType(StatementType stmtType) {
```

Comment:
See comment above. We probably don't want to set the type this way as it suggests that the type can change over time.